File: printPrimes.py
def printPrimes(n):
	primes = []
	for i in range(0,n+1):
		primes.append(True)
	divisor = 2
	while divisor*divisor<=n:
		if primes[divisor]==True:
			i = 2*divisor
			while i<=n:
				primes[i]=False
				i= i + divisor
	for i in range(2,n):
		if primes[i]==True:
			print(i)

File: hw/hw1/printPrimes.java
class printPrimes {
	public static void main main(String args[]){
		printPrimes.printPrimes(100)
	}
	public static void printPrimes(int n) {
      boolean[] prime = new boolean[n + 1];                  // Numbered 0...n.
      int i;
      for (i = 2; i <= n; i++) {
        prime[i] = true;                       // Prime until proven composite.
      }
      for (int divisor = 2; divisor * divisor <= n; divisor++) {
        if (prime[divisor]) {
          for (i = 2 * divisor; i <= n; i = i + divisor) {
            prime[i] = false;                     // i is divisible by divisor.
          }
        }
      }
      for (i = 2; i <= n; i++) {
        if (prime[i]) {
          System.out.print(" " + i);
        }
      }
    }
}